test/integration/integration_test.cc

```diff
@@ -1550,6 +1550,36 @@ TEST_P(IntegrationTest, TestUpgradeHeaderInResponse) {
   EXPECT_EQ("Hello World", response->body());
 }
 
+TEST_P(IntegrationTest, TestUpgradeHeaderInResponseWithTrailer) {
```

Comment:
Can you add a short `//` above this explaining what it demonstrates for posterity?